packages/workspace/src/workspace/nacl_files/nacl_file_update.ts

```diff
@@ -64,41 +73,107 @@ export const getChangeLocations = (
     }
   }
 
-  const findLocations = (): parser.SourceRange[] => {
+  const getFollowingElementsInParent = (): [ElemID[], number | undefined] => {
+    if (change.baseChange === undefined) {
+      log.warn(`No base change: ${change}`)
+      return [[], undefined]
+    }
+
+    const changeData = getChangeData(change)
+    const parent = changeData instanceof Field ? changeData.parent : getChangeData(change.baseChange)
+    const pathInParent = getPath(parent, change.id)
+    if (pathInParent === undefined) {
+      log.warn(`Could not get path for ${change.id} in parent: ${parent}`)
+      return [[], undefined]
+    }
+
+    const container = _.get(parent, pathInParent.slice(0, -1))
+    if (!(container instanceof Object)) {
+      log.warn(`Got non object container ${container} for path ${pathInParent.slice(0, -1)} in parent: ${parent}`)
+      return [[], undefined]
+    }
+
+    const idNameParts = change.id.getFullNameParts()
+    const elementName = idNameParts.pop() as string // Element can't have an empty ID
+    const index = Object.keys(container).indexOf(elementName)
+    if (index === -1) {
+      log.warn(`Element ${elementName} not found in container: ${container}`)
```

Comment:
nit - I tend to prefer string formatting over template strings (I do it because it allows the code to skip actually creating the log string if it is not going to be logged)

also, you shouldn't log element values directly.
elements values can contain reference expressions that could lead to a LOT of information that you do not want to log (makes to log potentially very large and hard to understand), we have a few overrides to make `inspect` output a more log friendly representation of values ([example](https://github.com/salto-io/salto/blob/main/packages/adapter-api/src/values.ts#L160))

```suggestion
      log.warn('Element %s not found in container: %s', elementName, inspectValue(container))
```

Reply:
Done.